snitun/multiplexer/core.py

```diff
@@ -312,17 +326,33 @@ async def _process_message(self, message: MultiplexerMessage) -> None:
                     MultiplexerMessage(message.id, CHANNEL_FLOW_PING, b"", b"pong"),
                 )
 
+        # Pause or Resume
+        elif message.flow_type in (CHANNEL_FLOW_PAUSE, CHANNEL_FLOW_RESUME):
+            # When the remote input is under water state changes
+            # call the on_remote_input_under_water method
+            under_water = message.flow_type == CHANNEL_FLOW_PAUSE
+            if channel_ := self._channels.get(message.id):
+                channel_.on_remote_input_under_water(under_water)
```

Comment:
Still needs a test that the remote receives this and acts on it

Should be end to end which means we need to patch the queue sizes quite low and block the input queue from being read